### PR TITLE
Convert `di` commands to the rzshell

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

- Convert `di` commands to the rzshell
- Rename `dif` to `diff` and move it into the "shell" commands category

**Test plan**

CI is green

**Closing issues**

Partially addresses https://github.com/rizinorg/rizin/issues/1342

### CODE REVIEW

File: librz/core/cmd/cmd_shell.c [hunk 1]
```diff
@@ -233,3 +233,33 @@ RZ_IPI RzCmdStatus rz_cmd_shell_fortune_handler(RzCore *core, int argc, const ch
...
+		return RZ_CMD_STATUS_ERROR;
+	}
+	char *b = (char *)getFileData(core, argv[2]);
+	if (!b) {
+		RZ_LOG_ERROR("core: Cannot open file B: \"%s\"\n", argv[2]);
+		free(a);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	RzDiff *dff = rz_diff_lines_new(a, b, NULL);
+	char *uni = rz_diff_unified_text(dff, argv[1], argv[2], false, false);
```
Comment (wargio) [inline librz/core/cmd/cmd_shell.c]: maybe add a `bool color = rz_config_i("scr.color") > 0` and use `color` as value for the last false

File: librz/core/cmd/cmd_shell.c [hunk 2]
```diff
@@ -233,3 +233,33 @@ RZ_IPI RzCmdStatus rz_cmd_shell_fortune_handler(RzCore *core, int argc, const ch
 	rz_core_fortune_print_random(core);
 	return RZ_CMD_STATUS_OK;
 }
+
+static ut8 *getFileData(RzCore *core, const char *arg) {
+	if (*arg == '$') {
+		return (ut8 *)rz_cmd_alias_get(core->rcmd, arg, 1);
```
Comment (wargio) [inline librz/core/cmd/cmd_shell.c]: shouldn't the alias be the argument of slurp ?

Comment (XVilka) [reply librz/core/cmd/cmd_shell.c]: I get that it compares the contents of two aliases, not sure why.

**Review by ret2libc — APPROVED** [review]

File: librz/core/cmd/cmd_shell.c (hunk 2 above)
Comment (wargio) [reply librz/core/cmd/cmd_shell.c]: if you read what that command used to do, it says that it use the value stored in the alias as the filename of the bin to diff

Comment (wargio) [reply librz/core/cmd/cmd_shell.c]: `"dif", " [$a] [$b]", "Compare two files (or $alias files)",`

Comment (wargio) [reply librz/core/cmd/cmd_shell.c]: i guess this was never tested. i'm ok to remove this feature and just diff 2 files.

Comment (XVilka) [reply librz/core/cmd/cmd_shell.c]: Removed alias diffing then.

**Pull request merged**